crates/shared/src/signature_validator/simulation.rs

```diff
@@ -18,18 +19,54 @@ pub struct Validator {
     signatures: contracts::support::Signatures,
     settlement: H160,
     vault_relayer: H160,
+    web3: Web3,
 }
 
 impl Validator {
+    /// The result returned from `isValidSignature` if the signature is correct
+    const IS_VALID_SIGNATURE_MAGIC_BYTES: &'static str = "1626ba7e";
+
     pub fn new(web3: &Web3, settlement: H160, vault_relayer: H160) -> Self {
         let web3 = ethrpc::instrumented::instrument_with_label(web3, "signatureValidation".into());
         Self {
             signatures: contracts::support::Signatures::at(&web3, settlement),
             settlement,
             vault_relayer,
+            web3: web3.clone(),
         }
     }
 
+    /// Simulate isValidSignature for the cases in which the order does not have
+    /// pre-interactions
+    async fn simulate_without_pre_interactions(
+        &self,
+        check: &SignatureCheck,
+    ) -> Result<Simulation, SignatureValidationError> {
+        // We simulate the signature verification from the owner contract's
```

Comment:
This comment does not apply to the simpler check.
IMO it would be more valuable to explain why a signature without pre-interactions doesn't need the full simulation.